Data/ByteString/Builder/RealFloat.hs

```diff
@@ -0,0 +1,227 @@
+-- |
+-- Module      : Data.ByteString.Builder.RealFloat
+-- Copyright   : (c) Lawrence Wu 2021
+-- License     : BSD-style
+-- Maintainer  : lawrencejwu@gmail.com
+--
+-- Floating point formatting for Bytestring.Builder
+--
+-- This module primarily exposes `floatDec` and `doubleDec` which do the
+-- equivalent of converting through `string7 . show`.
+--
+-- It also exposes `formatFloat` and `formatDouble` with a similar API as
+-- `GHC.Float.formatRealFloat`.
+--
+-- NB: this implementation matches `show`'s output (specifically with respect
+-- to default rounding and length). In particular, there are boundary cases
+-- where the closest and 'shortest' string representations are not used.
+-- Mentions of 'shortest' in the docs below are with this caveat.
+
+module Data.ByteString.Builder.RealFloat
```

Comment:
I wonder whether there's a better module name for this module. Is there a good motivation for naming it `D.B.B.RealFloat`? If it's about the instances of `RealFloat`, why not name it `D.B.B.RealFrac` instead?! `RealFrac` has basically the same instances after all.

Or more seriosly: `D.B.B.Floats`?!

Reply:
I think this is another artifact of `GHC.Float` and `format**RealFloat**`. Looking at the classes at least though, we do depend on `floatRange` and `floatDigits` from `RealFloat` (though these could be relatively trivially made constants as we have for several other float and double bitcounts)

Reply:
I'm fine with sticking with this name for now – I don't have a much better suggestion at least.

Thoughts, @Bodigrim?

Reply:
I'm fine with this name.